src/foam/nanos/boot/NSpecFactory.java

```diff
@@ -45,7 +48,14 @@ public synchronized Object create(X x) {
 
     try {
       if ( logger != null ) logger.info("Creating Service", spec_.getName());
-      ns_ = spec_.createService(x_.getX().put(NSpec.class, spec_));
+      if ( spec_.getLanguage() == foam.nanos.script.Language.BEANSHELL ) {
+        ns_ = spec_.createService(x_.getX().put(NSpec.class, spec_),null);
+      }
+      else if (spec_.getLanguage() == foam.nanos.script.Language.JSHELL) {
+        //PrintStream ps =null;
+        ns_ = spec_.createService(x_.getX().put(NSpec.class, spec_),null);
```

Comment:
there doesn't seem to be a difference between this line and the line above for beanshell .. so why differentiate between the two?